### DISCUSSION TITLE
How to get to know a property is locked or not?

### DISCUSSION THREAD

**Discussion by nagsamayam** [spatie/laravel-settings#20] [Q&A] [answered]
Thank you

**Comment by rubenvanassche** [comment, marked as answer]
Hi @nagsamayam,

I've just released version 1.0.4 which includes a `getLockedProperties` method!

**Reply by cjonstrup** [reply]
Did the getLockedProperties add an extra query?

select "name", "payload" from "settings" where "group" = 'general'
select "name" from "settings" where "group" = 'general' and "locked" = 1

**Reply by rubenvanassche** [reply]
Yes that's true since we need to go over the Repository to get the most up to date locked properties. Not sure if keeping these cached is a good idea.
